docs/markdown/snippets/pkgconfig_break.md

```diff
@@ -0,0 +1,26 @@
+## Behaviour change in pkg-config generator
+
+A small behaviour change has been made in the pkg-config generator. It will only
+generate a `Requires:` or `Requires.private:` for a `library()` if is it passed
+as first positional argument of the `generate()` method, and not any more if it
+is passed to the `libraries` keyword argument any more.
```

Comment:
I think this paragraph is hard to understand correctly.

We don't seem to have any wording at all about this important effect of associating internal libraries with the generated pkg-config files.

I think i would rephrase this as using the positional argument for the main library is now mandatory. That is very actionable for our users and we can then after this simple case add a caveat about previously allowing multiple libraries to be associated with one pkg-config file and that this will no longer be possible. (If we go with the depreciation idea i proposed it will still be possible for a time but we should urge users to get in contact with their use case anyway to see if we are missing something)

Reply:
Yes, of course. It's only mandatory for pkg-config files that represent built libraries. But i still think building around that would be more understandable. But if you come up with something else that's easy to understand that's fine too. But i don't like how the current one focuses on generated lines and not on something high level. Also i had a hard time understanding it and i worked on this code. Maybe it could also phrased something like "to allow further dependencies on a library for which a pkg-config file was generated with pkg.generate to refer to the generated file..."

Reply:
I tried to rephrase it, does it look better? I'm not really good at writing doc, tbh :(

Reply:
Changed this text to be a deprecation warning instead of behaviour change.